File: app/src/main/java/android/mahendra/attendancemanager/MainActivity.java
package android.mahendra.attendancemanager;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import android.os.Bundle;

public class MainActivity extends SingleFragmentActivity {

    @Override
    protected Fragment createFragment() {
        return SubjectListFragment.newInstance();
    }
}

File: app/src/main/java/android/mahendra/attendancemanager/activities/main/MainActivity.kt
package android.mahendra.attendancemanager.activities.main

import android.mahendra.attendancemanager.activities.SingleFragmentActivity
import android.mahendra.attendancemanager.activities.main.fragments.SubjectListFragment
import androidx.fragment.app.Fragment

class MainActivity : SingleFragmentActivity() {
    override fun createFragment(): Fragment? {
        return SubjectListFragment.newInstance()
    }
}